guides/v2.2/cloud/reference/discover-deploy.md

```diff
@@ -31,33 +29,101 @@ A set of YAML configuration files located in the project root directory define y
--  **Test your build and deploy locally and in Staging before deploying to Production.** Extensions and custom code work great in development. Some users push to production only to have failures and issues. Staging gives you an opportunity to fully test your code and implementation in a production environment without extended downtime if something goes wrong in Production.
+-  **Ensure that you are running the most current version of ece-tools**–See {{ site.data.var.ct }} [Release notes]({{ page.baseurl }}/cloud/release-notes/cloud-tools.html).
+
+-  **Follow the build and deploy process**–Ensure that you have the correct code in each environment to avoid overwriting configuration when merging code between environments. For example, to make configuration changes that apply to all environments, modify and test the changes in the local environment before deploying to Integration, and then deploy and test the changes in Staging before deploying to Production. When you merge from one environment to another, the deployment overwrites all code in the remote environment, except environment-specific configuration and settings. See [Build and deploy full steps](#steps).
+
+-  **Use the same variables across environments**–The values for these variables may differ across environments, but the same variables are usually required in each environment. See [Configuration management for store settings](/{{ page.baseurl }}//cloud/live/sens-data-over.html).
+
+-  **Keep sensitive configuration values and data in environment-specific variables**–This includes variables specified using the Magento Cloud CLI, the Project Web interface, or added to the `env.php` file. See [Working with environment variables]({{ page.baseurl }}/cloud/env/working-with-variables.html).
+
+-  **Ensure that all code is available in the environment branch**–If you point to or include hooks to code in other branches, such as a private branch, you may encounter problems with the build and deploy process. For example, if you include a theme from a private repository in your branch, the theme will not build with the Magento code.
+
+-  **Add new extensions, integrations, and code in iterated branches**–Make and test changes locally, push to Integration, then to Staging and Production. Test and resolve issues in each environment before merging the updates to the next environment. Some extensions and integrations must be enabled and configured in a specific order due to dependencies. Adding these in groups can make your build and deploy process much easier and help determine where issues occur.
+
+-  **Verify service versions and relationships and ability to connect as needed**–See [Service versions]({{ page.baseurl }}/cloud/project/project-conf-files_services.html#service-versions) and [Service relationships]({{ page.baseurl }}/cloud/project/project-conf-files_services.html#service-relationships).
```

Comment:
This may need a little more help.
```suggestion
-  **Verify service versions and relationships and the ability to connect**–Verify the services that are available to your application and ensure you are using the most appropriate version. See [Service versions]({{ page.baseurl }}/cloud/project/project-conf-files_services.html#service-versions) and [Service relationships]({{ page.baseurl }}/cloud/project/project-conf-files_services.html#service-relationships).
```